java/src/com/mongodb/csfle/CSFLEHelpers.java

```diff
@@ -0,0 +1,219 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+/*
+ * Helper methods and sample data for this companion project.
+ */
+public class CSFLEHelpers {
+
+    // Sample data
+    public static String SAMPLE_NAME_VALUE = "John Doe";
+    public static Integer SAMPLE_SSN_VALUE = 145014000;
+
+    public static Document SAMPLE_DOC = new Document()
+            .append("name", SAMPLE_NAME_VALUE)
+            .append("ssn", SAMPLE_SSN_VALUE)
+            .append("bloodType", "AB-")
+            .append("medicalRecords", Arrays.asList(
+                    new Document().append("weight", "180"),
+                    new Document().append("bloodPressure", "120/80")))
+            .append("insurance", new Document()
+                .append("policyNumber", 123142)
+                .append("provider", "MaestCare"));
+
+    // Reads the 96-byte local master key
+    public static byte[] readMasterKey(String filePath) throws IOException {
+        FileInputStream fis = null;
+        int numBytes = 96;
+        byte[] fileBytes = new byte[numBytes];
+
+        try {
+            fis = new FileInputStream(filePath);
+            if (fis.read(fileBytes) < numBytes)
+                throw new Exception("Expected to read 96 bytes from file");
+        } catch(Exception e) {
+            e.printStackTrace();
+            System.exit(1);;
+        } finally {
+            if (fis != null)
+                fis.close();
+        }
+        return fileBytes;
+    }
+
+    // JSON Schema helpers
+    private static String DETERMINISTIC_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Deterministic";
+    private static String RANDOM_ENCRYPTION_TYPE = "AEAD_AES_256_CBC_HMAC_SHA_512-Random";
+
+    private static Document buildEncryptedField(String keyId, String bsonType, Boolean isDeterministic) {
+        return new Document().
+                append("encrypt", new Document()
+                        .append("bsonType", bsonType)
+                        .append("algorithm",
+                                (isDeterministic) ? DETERMINISTIC_ENCRYPTION_TYPE : RANDOM_ENCRYPTION_TYPE));
+    }
+
+    private static Document createEncryptMetadataSchema(String keyId) {
+        List<Document> keyIds = new ArrayList<Document>();
+        keyIds.add(new Document()
+                .append("$binary", new Document()
+                        .append("base64", keyId)
+                        .append("subType", "04")));
+        return new Document().append("keyId", keyIds);
+    }
+
+    public static Document createJSONSchema(String keyId) throws IllegalArgumentException {
+        if (keyId.isEmpty()) {
+            throw new IllegalArgumentException("keyId must contain your base64 encryption key id.");
+        }
+        return new Document().append("bsonType", "object").append("encryptMetadata", createEncryptMetadataSchema(keyId))
```

Comment:
For the schema, consider just using a multi-line string of JSON, and use Document.parse to turn it into a Document.  Makes it a bit more readable.

Reply:
Thanks, readability is a valid concern. Is the multi-line JSON string the "canonical" way? This currently matches the code in our CSFLE guide so this PR will involve updating it in both locations.

Could we update it later once there is a canonical way so that we do not have to potentially update it twice?

Reply:
I wouldn't go that far as to say it's canonical, as we don't have a lot of experience of users actually taking advantage of JSON schema.